nservicebus/outbox/index.md

```diff
@@ -79,8 +79,61 @@ In order to gradually convert an entire system from the DTC to the Outbox:
+
+
+## Concurrency
+
+Outbox cannot guarantee that handlers or sagas are invoked exactly once. Outbox does not prevent concurrent attempts to process the same message but ensures that the outcome of the processing is persisted only once for resources that share the outbox transaction.
+
+The following scenarios might cause handlers to be invoked multiple times:
+
+- Sender sends multiple physical message with the same message identifier (submit order form)
+- Transport can send multiple messages when having a failover, they operate in a at-least-once delivery model
+- The outbox dispatch stage can be invoked more-than-once if the previous dispatch (partially) failed and is retried.
+
+In those scenarios, multiple physical copies of the same logical message exist in the queue. If an endpoint allows concurrent message processing or is scaled out it can happen that these multiple copies are processed concurrently.
+
+
+## Non transactional resources
+
+If non-transactional resources are being accessed then make sure processing is idempotent. Non-transactional resources are not guaranteed to be handled in an idempotent way by the Outbox. If more-than-once invocation is unwanted then this can be almost prevented completely by making sure only a single endpoint instance is running and have [configured the endpoint to process messages sequentially by setting the maximum concurrency to 1](/nservicebus/operations/tuning.md#tuning-concurrency). Only data modified via a shared storage session is made idempotent via the outbox and has exact-once semantics. See [persistence specific documention](/nservicebus/outbox/#Persistence) on how to obtain the shared storage session.
+
+NOTE: This problem is not specific to Outbox but mentioned here as sometimes it is assumed Outbox does deduplication that prevents more-than-once invocation.
+
+Try to avoid mixing transactional and non transactional tasks. If performing non transactional tasks send a message to perform this task in isolation.
+
+
+## Outbox expiration duration
+
+Part of the outbox is to guarantee that the data remains consistent if the same message is processed more-than-once. A message with the exact same identification can be detected and ignored. This data is stored in the outbox record and kept for a configurable duration. The default expiration duration depends per persistence implementation. All implemetations can be configured on how long an outbox record needs to be stored and some can be configured on how often outbox cleanup needs to occur.
```

Comment:
> Part of the outbox is to guarantee....

should this be?
> Part of the purpose of the outbox is to guarantee....

Reply:
No need to "dash" `more-than-once` in the first sentence.

Reply:
> This data is stored in the outbox record

Which data? Use something like "The message indentification data is stored..." instead of the word "this" to add clarity

Reply:
> This data is stored in the outbox record and kept for a configurable duration. The default expiration duration depends per persistence implementation. All implemetations can be configured on how long an outbox record needs to be stored and some can be configured on how often outbox cleanup needs to occur.

would be better as
> The identification data for each outbox record is retained. The default duration that the data is retained for will vary depending on the persistence chosen for the outbox. That default duration, as well as the frequency of data removal, can be overrode for all outbox persistences.